### PR TITLE
[7.67.x-blue] RHPAM-3709: upgrade maven dependencies to address CVE-2021-26291

### PR DESCRIPTION
**Pull request by baldimir**
Replaces https://github.com/kiegroup/jbpm/pull/2398

### CODE REVIEW

**Comment by baldimir** [comment]
jenkins retest this

**Comment by mareknovotny** [comment]
could it be here some problems with transitive deps of ShrinkWrap? 
```
java.lang.RuntimeException: Could not invoke deployment method: public static org.jboss.shrinkwrap.api.Archive org.jbpm.test.container.AbstractEJBTransactionsTest.deployEJBTest()
Caused by: java.lang.reflect.InvocationTargetException
Caused by: org.jboss.shrinkwrap.resolver.api.Invokable$InvocationException: Unable to invoke onlyOne([Ljava.lang.Class;@7f9e1534) on object org.jboss.shrinkwrap.resolver.spi.loader.ServiceRegistry with parameters [Ljava.lang.Object;@138a7441
Caused by: java.lang.reflect.InvocationTargetException
Caused by: java.lang.RuntimeException: Could not create new service instance
Caused by: java.lang.reflect.InvocationTargetException
Caused by: java.lang.NoClassDefFoundError: org/eclipse/aether/internal/impl/DefaultDependencyCollector
```
in https://github.com/baldimir/jbpm/blob/RHPAM-3709/jbpm-container-test/jbpm-in-container-test/jbpm-container-test-suite/pom.xml#L192-L231 ?

**Comment by elguardian** [comment]
skinwrap depends on maven resolver impl that depens on aehter. this is not included in the pom. 
I don't see how you want to workaround this 
@baldimir

File: jbpm-container-test/jbpm-in-container-test/jbpm-container-test-suite/pom.xml [hunk 1]
```diff
@@ -35,7 +35,9 @@
     <container.log.file>${container.profile.dir}/profile-log.txt</container.log.file>
 
     <!-- Overridden until jboss-integration-platform-parent is upgraded -->
-    <version.org.jboss.shrinkwrap.resolver>3.1.0</version.org.jboss.shrinkwrap.resolver>
+    <version.org.jboss.shrinkwrap.resolver>3.3.0</version.org.jboss.shrinkwrap.resolver>
+    <version.org.apache.maven>3.9.3</version.org.apache.maven>
+    <version.org.apache.maven.resolver>1.9.13</version.org.apache.maven.resolver>
```
Comment (elguardian) [inline jbpm-container-test/jbpm-in-container-test/jbpm-container-test-suite/pom.xml:40]: missed guava
<version.com.google.guava>20.0</version.com.google.guava>

File: jbpm-container-test/jbpm-in-container-test/jbpm-container-test-suite/pom.xml [hunk 2]
```diff
@@ -38,6 +38,7 @@
     <version.org.jboss.shrinkwrap.resolver>3.3.0</version.org.jboss.shrinkwrap.resolver>
     <version.org.apache.maven>3.9.3</version.org.apache.maven>
     <version.org.apache.maven.resolver>1.9.13</version.org.apache.maven.resolver>
+    <version.com.google.guava>20.0</version.com.google.guava>
```
Comment (mareknovotny) [inline jbpm-container-test/jbpm-in-container-test/jbpm-container-test-suite/pom.xml:41]: guava 20? this version should not be used (CVEs https://www.cvedetails.com/vulnerability-list/vendor_id-1224/product_id-52274/Google-Guava.html)

Comment (elguardian) [reply jbpm-container-test/jbpm-in-container-test/jbpm-container-test-suite/pom.xml:41]: let see if it works for 30 but the curren shrinkwrap needs some alignments.

Comment (baldimir) [reply jbpm-container-test/jbpm-in-container-test/jbpm-container-test-suite/pom.xml]: I updated with 33.1.0-jre, which is the latest one and it looks like working for me locally. Let's see in the PR checks.

**Comment by mareknovotny** [comment]
Jenkins retest this please

**Comment by mareknovotny** [comment]
ok, now there are only 3 failed tests in comparison to 16 failures previously.

All 3 have the same root cause:

```
java.lang.RuntimeException: Could not invoke deployment method: public static org.jboss.shrinkwrap.api.Archive org.jbpm.test.container.test.WebServiceInvokeTest.deployWebService()
	at org.jbpm.test.container.test.WebServiceInvokeTest.deployWebService(WebServiceInvokeTest.java:49)
Caused by: java.lang.reflect.InvocationTargetException
	at org.jbpm.test.container.test.WebServiceInvokeTest.deployWebService(WebServiceInvokeTest.java:49)
Caused by: java.lang.RuntimeException: Could not create new service instance
	at org.jbpm.test.container.test.WebServiceInvokeTest.deployWebService(WebServiceInvokeTest.java:49)
Caused by: java.lang.reflect.InvocationTargetException
	at org.jbpm.test.container.test.WebServiceInvokeTest.deployWebService(WebServiceInvokeTest.java:49)
Caused by: com.google.inject.ProvisionException: 
Unable to provision, see the following errors:

1) Error in custom provider, com.google.inject.OutOfScopeException: Cannot access session scope outside of a scoping block
  at ClassRealm[maven, parent: ClassRealm[maven-parent, parent: null]] (via modules: org.eclipse.sisu.wire.WireModule -> org.eclipse.sisu.plexus.PlexusBindingModule -> org.apache.maven.session.scope.internal.SessionScopeModule)
  while locating org.apache.maven.execution.MavenSession

1 error
	at org.jbpm.test.container.test.WebServiceInvokeTest.deployWebService(WebServiceInvokeTest.java:49)
Caused by: com.google.inject.OutOfScopeException: Cannot access session scope outside of a scoping block
	at org.jbpm.test.container.test.WebServiceInvokeTest.deployWebService(WebServiceInvokeTest.java:49)
```

**Comment by elguardian** [comment]
* some notes in here. The error is a bit random and happens if you execute all the test in the suite. In local I managed to get 2 of them.

[ERROR] Errors: 
[ERROR]   RestServiceInvokeTest.org.jbpm.test.container.test.RestServiceInvokeTest » Runtime
[ERROR]   TransactionsTest.org.jbpm.test.container.test.TransactionsTest » Runtime Could...

seems like an error in the clean up

**Comment by elguardian** [comment]
Tested in my local seems a class loading problem
in failsafe
<reuseForks>false</reuseForks>
<forkCount>1</forkCount>
<useSystemClassLoader>false</useSystemClassLoader>

**Review by mareknovotny — APPROVED** [review]

**Pull request merged**